src/jdk.incubator.vector/share/classes/jdk/incubator/vector/X-Vector.java.template

```diff
@@ -821,6 +821,39 @@ public abstract class $abstractvectortype$ extends AbstractVector<$Boxtype$> {
                     v0.uOp(m, (i, a) -> ($type$) Math.expm1(a));
             case VECTOR_OP_LOG1P: return (v0, m) ->
                     v0.uOp(m, (i, a) -> ($type$) Math.log1p(a));
+#else[!short]
+            case VECTOR_OP_SIN: return (v0, m) ->
+                    v0.uOp(m, (i, a) -> Halffloat.valueOf((float) Math.sin(Halffloat.valueOf(a).floatValue())));
+            case VECTOR_OP_COS: return (v0, m) ->
+                    v0.uOp(m, (i, a) -> Halffloat.valueOf((float) Math.cos(Halffloat.valueOf(a).floatValue())));
+            case VECTOR_OP_TAN: return (v0, m) ->
+                    v0.uOp(m, (i, a) -> Halffloat.valueOf((float) Math.tan(Halffloat.valueOf(a).floatValue())));
+            case VECTOR_OP_ASIN: return (v0, m) ->
+                    v0.uOp(m, (i, a) -> Halffloat.valueOf((float) Math.asin(Halffloat.valueOf(a).floatValue())));
+            case VECTOR_OP_ACOS: return (v0, m) ->
+                    v0.uOp(m, (i, a) -> Halffloat.valueOf((float) Math.acos(Halffloat.valueOf(a).floatValue())));
+            case VECTOR_OP_ATAN: return (v0, m) ->
+                    v0.uOp(m, (i, a) -> Halffloat.valueOf((float) Math.atan(Halffloat.valueOf(a).floatValue())));
+            case VECTOR_OP_EXP: return (v0, m) ->
+                    v0.uOp(m, (i, a) -> Halffloat.valueOf((float) Math.exp(Halffloat.valueOf(a).floatValue())));
+            case VECTOR_OP_LOG: return (v0, m) ->
+                    v0.uOp(m, (i, a) -> Halffloat.valueOf((float) Math.log(Halffloat.valueOf(a).floatValue())));
+            case VECTOR_OP_LOG10: return (v0, m) ->
+                    v0.uOp(m, (i, a) -> Halffloat.valueOf((float) Math.log10(Halffloat.valueOf(a).floatValue())));
+            case VECTOR_OP_SQRT: return (v0, m) ->
+                    v0.uOp(m, (i, a) -> Halffloat.valueOf((float) Math.sqrt(Halffloat.valueOf(a).floatValue())));
+            case VECTOR_OP_CBRT: return (v0, m) ->
+                    v0.uOp(m, (i, a) -> Halffloat.valueOf((float) Math.cbrt(Halffloat.valueOf(a).floatValue())));
+            case VECTOR_OP_SINH: return (v0, m) ->
+                    v0.uOp(m, (i, a) -> Halffloat.valueOf((float) Math.sinh(Halffloat.valueOf(a).floatValue())));
+            case VECTOR_OP_COSH: return (v0, m) ->
+                    v0.uOp(m, (i, a) -> Halffloat.valueOf((float) Math.cosh(Halffloat.valueOf(a).floatValue())));
+            case VECTOR_OP_TANH: return (v0, m) ->
+                    v0.uOp(m, (i, a) -> Halffloat.valueOf((float) Math.tanh(Halffloat.valueOf(a).floatValue())));
```

Comment:
All the Math.* transcendental routines accept a double argument, you can directly use Float.float16ToFloat() instead of Halffloat.valueOf(a).floatValue(), argument will get upcasted to double.

Reply:
A separate patch is there to replace at all places Halffloat.valueOf(a).floatValue() with Float.float16ToFloat() , Should I do the replacement changes in this patch at all places.

Reply:
I do not see any harm in updating this patch with suggested changes, for existing checked in code a separate patch can be released.